### PR TITLE
ENH: Allow exponentially weighted functions to specify alpha directly

### LINKED ISSUE #10789: ENH: Allow exponential weighing functions to specify alpha, in addition to span / com / halflife
Currently the exponential functions take one of three arguments to specify the length of backhistory. Each of these numerically converts to alpha. Outlined here: http://pandas.pydata.org/pandas-docs/stable/computation.html#exponentially-weighted-moment-functions

Is there a reason we don't allow users to just pass in `alpha`? This is how I think about weighing some of the time.

I'm happy to do a PR in time; this issue is to solicit feedback.

### PR DESCRIPTION
**Pull request by evectant**
- Closes #10789
- Passes Travis CI, nosetests, and flake8
- Covered by new tests in test_window.py:TestMoments
- Documented in computation.rst and v0.18.0.txt

### CODE REVIEW

**Comment by jreback** [comment]
I am not sure we do any validity checks on any com parameters

eg is alpha=0 valid?

**Comment by evectant** [comment]
That's actually a question I had. Zero alpha is not okay, but zero half-life isn't either, and the code is not checking for that. Should I add the checks for `0 < alpha <= 1` and `span >= 1`, `com >= 0`, `half-life > 0` following from that?

**Comment by jreback** [comment]
yep I think we need some sanity checks as well here - raising a helpful ValueError would be nice (rather than giving weird/odd results)

File: pandas/tests/test_window.py [hunk 1]
```diff
@@ -1066,6 +1066,67 @@ def test_ewma_halflife_arg(self):
                               halflife=50)
             self.assertRaises(Exception, mom.ewma, self.arr)
 
+    def test_ewma_alpha_old_api(self):
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
```
Comment (jreback) [inline pandas/tests/test_window.py]: add the issue number as a comment here

**Comment by jreback** [comment]
thanks @evectant just minor comments.
ping when pushed and green.

**Comment by evectant** [comment]
@jreback: Added the comments, slightly reworded the description in computation.rst. Pushed and green.

File: pandas/core/window.py [hunk 2]
```diff
@@ -1013,12 +1013,16 @@ class EWM(_Rolling):
...
     span : float, optional
-        Specify decay in terms of span, :math:`\alpha = 2 / (span + 1)`
+        Specify decay in terms of span,
+        :math:`\alpha = 2 / (span + 1)`
     halflife : float, optional
-        Specify decay in terms of halflife,
+        Specify decay in terms of half-life,
         :math:`\alpha = 1 - exp(log(0.5) / halflife)`
+    alpha : float, optional
+        Specify smoothing factor :math:`\alpha` directly
```
Comment (jreback) [inline pandas/core/window.py]: can you add a versionadded tag here

File: doc/source/computation.rst [hunk 3]
```diff
@@ -743,14 +743,15 @@ directly, it's easier to think about either the **span**, **center of mass
...
+        1 - \exp^{\frac{\log 0.5}{h}}, & h = \text{half-life}
     \end{cases}
 
-One must specify precisely one of the three to the EW functions. **Span**
-corresponds to what is commonly called a "20-day EW moving average" for
-example. **Center of mass** has a more physical interpretation. For example,
-**span** = 20 corresponds to **com** = 9.5. **Halflife** is the period of
-time for the exponential weight to reduce to one half.
+One must specify precisely one of **span**, **center of mass**, **half-life**
+and **alpha** to the EW functions. **Span** corresponds to what is commonly
```
Comment (jreback) [inline doc/source/computation.rst]: add somewhere that ``alpha`` can be specified starting in 0.18.0

File: doc/source/computation.rst [hunk 4]
```diff
@@ -743,14 +743,15 @@ directly, it's easier to think about either the **span**, **center of mass
...
+called an "N-day EW moving average". **Center of mass** has a more physical
+interpretation and can be thought of in terms of span: :math:`c = (s - 1) / 2`.
+**Half-life** is the period of time for the exponential weight to reduce to
```
Comment (jreback) [inline doc/source/computation.rst]: can you add a mini chart showing the allowed domains (use csv-table to show it in markdown)

File: pandas/core/window.py [hunk 5]
```diff
@@ -1037,7 +1041,7 @@ class EWM(_Rolling):
 
     Notes
     -----
-    Either center of mass, span or halflife must be specified
+    Either center of mass, span, half-life or alpha must be specified
 
     EWMA is sometimes specified using a "span" parameter `s`, we have that the
     decay parameter :math:`\alpha` is related to the span as
```
Comment (jreback) [inline pandas/core/window.py]: can you add the domain requirements here as well (after the formula is good)

File: pandas/stats/moments.py [hunk 6]
```diff
@@ -68,12 +68,16 @@
 
 
 _ewm_kw = r"""com : float. optional
-    Center of mass: :math:`\alpha = 1 / (1 + com)`,
+    Specify decay in terms of center of mass,
+    :math:`\alpha = 1 / (1 + com)`
 span : float, optional
```
Comment (jreback) [inline pandas/stats/moments.py]: add the domain here as well

**Comment by jreback** [comment]
looks good. just some minor documentation comments.

**Comment by jreback** [comment]
pls rebase and push again, master has been updated

File: pandas/core/window.py [hunk 7]
```diff
@@ -1294,20 +1297,32 @@ def dataframe_from_int_dict(data, frame_template):
...
-    valid_count = len([x for x in [com, span, halflife] if x is not None])
+def _get_center_of_mass(com, span, halflife, alpha):
+    valid_count = len([x for x in [com, span, halflife, alpha]
+                      if x is not None])
     if valid_count > 1:
-        raise Exception("com, span, and halflife are mutually exclusive")
-
-    if span is not None:
-        # convert span to center of mass
+        raise Exception("com, span, halflife, and alpha "
```
Comment (jreback) [inline pandas/core/window.py]: can you make this a ``ValueError``

File: pandas/core/window.py [hunk 8]
```diff
@@ -1294,20 +1297,32 @@ def dataframe_from_int_dict(data, frame_template):
...
         decay = 1 - np.exp(np.log(0.5) / halflife)
         com = 1 / decay - 1
-    elif com is None:
-        raise Exception("Must pass one of com, span, or halflife")
+    elif alpha is not None:
+        if alpha <= 0 or alpha > 1:
+            raise ValueError("alpha must satisfy: 0 < alpha <= 1")
+        com = (1.0 - alpha) / alpha
+    else:
+        raise Exception("Must pass one of com, span, halflife, or alpha")
```
Comment (jreback) [inline pandas/core/window.py]: same here

**Comment by evectant** [comment]
@jreback: Pushed and green.

**Comment by jreback** [comment]
thanks @evectant great PR!

**Comment by jreback** [comment]
give a look at the built documentation (should be done shortly), http://pandas-docs.github.io/pandas-docs-travis/ and make sure looks ok (obviously where you changed things!). If not pls issue a follow.

**Comment by evectant** [comment]
@jreback The docs look good to me. Thanks for your help with the commit.

**Comment by jreback** [comment]
looks good

thanks for the PR!

**Pull request closed without merging**